app/models/conversation.rb

```diff
@@ -0,0 +1,13 @@
+class Conversation < ActiveRecord::Base
+ belongs_to :sender, :foreign_key => :sender_id, class_name: ‘User’
+ belongs_to :recipient, :foreign_key => :recipient_id, class_name: ‘User’
+
+has_many :messages, dependent: :destroy
+validates_uniqueness_of :sender_id, :scope => :recipient_id
+scope :involving, -> (user) do
+ where("conversations.sender_id =? OR conversations.recipient_id  =?", user.id, user.id)
+ end
```

Comment:
`end` at 9, 1 is not aligned with `-> (user) do` at 7, 18 or `scope :involving, -> (user) do` at 7, 0